x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -62,24 +64,125 @@ export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isD
+      if (conn.isTechPreview) {
+        _append.push(
+          <EuiBadge key={key + '-preview'} iconType="beaker" color="hollow">
+            {i18n.translate(
+              'xpack.serverlessSearch.connectors.chooseConnectorSelectable.thechPreviewBadgeLabel',
+              { defaultMessage: 'Tech preview' }
+            )}
+          </EuiBadge>
+        );
+      }
+      if (conn.isBeta) {
+        _append.push(
+          <EuiBadge key={key + '-beta'} iconType={'beta'} color="hollow">
+            {i18n.translate(
+              'xpack.serverlessSearch.connectors.chooseConnectorSelectable.BetaBadgeLabel',
+              {
+                defaultMessage: 'Beta',
+              }
+            )}
+          </EuiBadge>
+        );
+      }
+      return {
+        _append,
+        _prepend: <EuiIcon size="l" type={conn.iconPath} />,
+        key: key.toString(),
+        label: conn.name,
+        serviceType: conn.serviceType,
+      };
+    });
+  };
+
+  const initialOptions = getInitialOptions();
+  const { euiTheme } = useEuiTheme();
+  const [selectedOption, setSelectedOption] = useState<Array<EuiComboBoxOptionOption<OptionData>>>(
+    []
+  );
+  const [selectedConnector, setSelectedConnector] = useState<Connector | null>(null);
+  const renderOption = (
+    option: EuiComboBoxOptionOption<OptionData>,
+    searchValue: string,
+    contentClassName: string
+  ) => {
+    const { _append, key, label, _prepend, serviceType } =
+      option as EuiComboBoxOptionOption<OptionData> & {
+        _append: JSX.Element[];
+        _prepend: JSX.Element;
+        serviceType: string;
+      };
+    return (
+      <EuiFlexGroup
+        gutterSize="m"
+        key={key + '-span'}
+        justifyContent="spaceBetween"
+        className={contentClassName}
+      >
+        <EuiFlexGroup gutterSize="m">
+          <EuiFlexItem grow={false}>{_prepend}</EuiFlexItem>
+          <EuiFlexItem
+            grow={false}
+            data-test-subj={`serverlessSearchConnectorServiceType-${serviceType}`}
+          >
+            <EuiText size="s" textAlign="left">
+              {label}
+            </EuiText>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiFlexItem grow={false}>{_append}</EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  };
+
   return (
-    <EuiFormRow
-      label={i18n.translate('xpack.serverlessSearch.connectors.serviceTypeLabel', {
-        defaultMessage: 'Connector type',
-      })}
-      data-test-subj="serverlessSearchEditConnectorType"
-      fullWidth
-    >
-      <EuiSuperSelect
-        // We only want to allow people to set the service type once to avoid weird conflicts
-        disabled={Boolean(connector.service_type) || isDisabled}
-        data-test-subj="serverlessSearchEditConnectorTypeChoices"
-        isLoading={isLoading}
-        onChange={(event) => mutate(event)}
-        options={options}
-        valueOfSelected={connector.service_type || undefined}
+    <>
```

Comment:
we're not rendering anything with the `<EuiFormRow>` so isn't this `<>` redundant?

Reply:
You're right. I forgot to clean this one while I was making exploration.